### PR TITLE
Web responsive menu improvements

### PR DESCRIPTION
**Pull request by danieltorres7**
Added apptitle before oe_submenu
Increased user avatar size to maximum
Fullscren dropdown menus on mobile mode

### CODE REVIEW

**Comment by pedrobaeza** [comment]
There are conflicts in the branch

**Comment by danieltorres7** [comment]
@pedrobaeza, all conflicts solved.

File: web_responsive/static/src/less/main.less
Comment (pedrobaeza) [inline web_responsive/static/src/less/main.less]: Please don't use important, as it overwrites everything else. Use a more specific selector instead.

**Comment by pedrobaeza** [comment]
In general, you must avoid to use `!important` in CSS styling

**Comment by danieltorres7** [comment]
@pedrobaeza. Succesfully replaced all important decorators by more specific selectors.

**Comment by danieltorres7** [comment]
Check it out @rousseldenis, @lasley and @leorochael. Everything seems to be working with the new menus.

**Review by pedrobaeza — APPROVED** [review]
Tested on runbot

**Comment by pedrobaeza** [comment]
@Yajo, can you review technically?

**Comment by danieltorres7** [comment]
I've completely rewritten the commits, removing redundant stuff and adding better comments and commit messages, but the behavior should remain the same (save for a small animation bug that I fixed).

File: web_responsive/static/src/less/navbar.less
Comment (lasley) [inline web_responsive/static/src/less/navbar.less]: Should we maybe use the Bootstrap vars here for consistency? - https://github.com/odoo/odoo/blob/10.0/addons/web/static/lib/bootstrap/less/variables.less#L361

File: web_responsive/static/src/less/navbar.less
Comment (yajo) [inline web_responsive/static/src/less/navbar.less]: I think you'd better use the corresponding [Odoo](https://github.com/odoo/odoo/blob/10.0/addons/web/static/src/less/variables.less) or [Bootstrap](https://github.com/odoo/odoo/blob/10.0/addons/web/static/lib/bootstrap/less/variables.less) variables for color and background-color.

File: web_responsive/static/src/less/navbar.less
Comment (yajo) [inline web_responsive/static/src/less/navbar.less]: `@screen-sm` is deprecated. Use `@screen-sm-min`.

File: web_responsive/static/src/less/navbar.less
Comment (yajo) [inline web_responsive/static/src/less/navbar.less]: Deprecated.

File: web_responsive/static/src/less/navbar.less
Comment (yajo) [inline web_responsive/static/src/less/navbar.less]: Deprecated.

File: web_responsive/static/src/less/navbar.less
Comment (yajo) [inline web_responsive/static/src/less/navbar.less]: Deprecated.

File: web_responsive/static/src/less/navbar.less
Comment (yajo) [inline web_responsive/static/src/less/navbar.less]: Again does any available variable fit in?

**Review by yajo — CHANGES_REQUESTED** [review]
Besides some technical details below, I think the top bar should be thick in sm size like it is in xs size, to make it more usable in tablets.

**Comment by danieltorres7** [comment]
@Yajo and @pedrobaeza. Requested changes implemented.

**Comment by danieltorres7** [comment]
> Besides some technical details below, I think the top bar should be thick in sm size like it is in xs size, to make it more usable in tablets.

@Yajo, I agree on you, anyway, how can we make thick top bars for tablets not changing desktop ones? I mean, resolution is higher than 768px on both and there is not available decorators (at least I don't know at the moment) to easily check if the device is a tablet.

**Comment by yajo** [comment]
There must be a media query that makes the bar thick in xs screens (or thin in >=sm). All you have to do is find it and change the width variable from xs to sm.

Oh, and about tablet or small PC, as long as tablet UI is usable in a small PC, you shouldn't care about desktop. At least until the [pointer media query](https://developer.mozilla.org/en/docs/Web/CSS/@media/pointer) lands mainstream.

**Comment by danieltorres7** [comment]
Thick top menus are already available for all sm screens, not only xs.

**Comment by leorochael** [comment]
@Yajo, do you want the top menubar to remain thick, no matter how large the screen?

This way the only difference between larger and smaller screens would be the presence of the menus directly on the top bar, or inside the menu button.

Is this what you want?

**Comment by danieltorres7** [comment]
I will refactor changes, so top menu height will remain the same on all modes, is that what you want? Always thick no matter what mode is active?

**Comment by pedrobaeza** [comment]
He will answer tomorrow, as he works only up to 15:00 CET

**Comment by lasley** [comment]
> Always thick no matter what mode is active?

@danieltorres7 @leorochael - I think he is saying always thick when on XS and SM, but leave it for the others. I think I agree, that would be a great change.

**Review by rousseldenis — APPROVED** [review]
:+1: Tested on runbot

File: web_responsive/static/src/less/navbar.less
Comment (lasley) [inline web_responsive/static/src/less/navbar.less]: Somewhat nitpicky, but maybe a `cursor: default` here to override `text`

**Review by lasley — APPROVED** [review]
@danieltorres7 Excellent improvement, few comments but otherwise 👍 

Looking at this, the navbar changing @Yajo is referring to is controlled with the [navbar-collapse, navbar-toggle](https://github.com/OCA/web/blob/10.0/web_responsive/views/web.xml#L43) classes.

I think maybe we should add this one to the roadmap. It's going to be getting into changing the way Bootstrap works, so I think it'll cause some cascading changes that'll blow this PR into a rabbit hole.

@Yajo - correct me if I'm wrong on my inferences.

**Review by yajo — DISMISSED** [review]
I'm afrraid not, friend. Just open the runbot and resize the window to sm size (768-991px width) and you'll see the top bar is thinner than xs (<=767px).

In any case, I'm not blocking this for that reason, but if you want to push that, it would be great 😉 

Note to PSC: **squash** at merge.

**Review by yajo — APPROVED** [review]
OK now I have been able to review properly. Indeed I meant what @lasley says. This looks good to me, but I'll wait for @danieltorres7 feedback before merging: do you want to address those changes @lasley said before we merge?

And about the thick bar, if it is too complicated, then maybe just a roadmap note would work for now.

Thanks!

**Comment by leorochael** [comment]
Ok, but considering that we can't tell large tablets apart from desktops before the pointer media query lands, wouldn't it be better (and simpler in the code) to just keep a wide top bar, no matter the resolution?

That's what Odoo does by default, BTW

**Comment by danieltorres7** [comment]
Top menu now is always thick, simplified some styles for that.

File: web_responsive/static/src/less/main.less
Comment (yajo) [inline web_responsive/static/src/less/main.less]: You knew this was going to happen... Please change this 😆 

A simple trick for these kind of overriding is:

```
html & {
  [syles...]
}
```

`html body` is more concise than `body` selector, so it will win the fight 😉

**Review by yajo — CHANGES_REQUESTED** [review]
You know what? It looks awesome. Just one more little change to merge 🎉

File: web_responsive/static/src/less/main.less
Comment (leorochael) [reply web_responsive/static/src/less/main.less]: What @danieltorres7 mean by "base template", is that there's a `style="overflow: auto;"` in the `<body>` tag ` *directly in the DOM*.

There's no way to add a more specific selector without `!important`

Comment (leorochael) [reply web_responsive/static/src/less/main.less]: Apparently this `style` attribute is being added by The `drawer.css` library because `<body>` has a `drawer` class, added by the `web_responsive` module. We're still investigating.

Comment (leorochael) [reply web_responsive/static/src/less/main.less]: This happens on the `drawer close` event of the app drawer, which happens automatically on the closing of the app drawer, or on window resize (triggers the drawer close).

At this point we would help suggestions on how to deal with the issue and how to remove the `!important`.

Comment (danieltorres7) [reply web_responsive/static/src/less/main.less]: Just removed overflow: hidden!important from main.less. Library drawer.3.2.0.js adds "overflow: auto" on close, also triggered on resize like @leorochael told us. This causes body to have it's own scrollbar and at small height screens we got 2 scrollbars at the same time. Committing that change on lib separately now...

File: web_responsive/static/lib/js/drawer.3.2.0.js
Comment (leorochael) [inline web_responsive/static/lib/js/drawer.3.2.0.js]: Given [@Yajo's comment on 511](https://github.com/OCA/web/pull/511#issuecomment-269595936), This line should probably be just removed, and the following block added to `web_responsive/static/lib/css/drawer.3.2.0.css`:

    .drawer-open {
        overlay: hidden;
    }

After doing the same in a PR for drawer, of course (which means, I should probably [update my own PR](https://github.com/blivesta/drawer/pull/36)...)

**Comment by danieltorres7** [comment]
@Yajo, requested changes and final fixes applied. We are ready for the final review.

**Comment by leorochael** [comment]
Please, let us know when it's ok to squash/reorder the commits.

**Comment by lasley** [comment]
@Yajo - the changes look good to you? 

@leorochael - Once he confirms, you're good to squash/reorder.

**Comment by rousseldenis** [comment]
Come back from holidays and changes seem great !
Thanks for the job !

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Please squash to merge 😉

**Comment by pedrobaeza** [comment]
I have squashed on merge.

**Comment by pedrobaeza** [comment]
Can you please backport this to v9 or isn't direct?

**Pull request merged**
